### PR TITLE
Allow custom local pow difficulty

### LINKED ISSUE #273: Add an option to specify multiplier when using local PoW
To let the user prioritize transactions. Only for CPU, GPU, and possibly future p2PoW

### PR DESCRIPTION
**Pull request by Joohansson**
- Only for CPU / GPU pow
- Overrides default SEND & RECEIVE (even if receive is 1/64x)
- Display pow notification until work is finished
- Can take VERY long time above 4x (GPU pow possibly never finish when 32x for example?)
- Resets local work cache when increasing multiplier (if user approve)

Fixes #273 

image

image

### CODE REVIEW

**Review by keerifox — APPROVED** [review]
seems to be working fine, code also looks ok

File: src/app/components/configure-app/configure-app.component.ts [hunk 1]
```diff
@@ -104,12 +104,24 @@ export class ConfigureAppComponent implements OnInit {
 
   powOptions = [
     { name: 'Best Option Available', value: 'best' },
-    { name: 'Client Side - WebGL [Recommended] (Chrome/Firefox)', value: 'clientWebGL' },
-    { name: 'Client Side - CPU', value: 'clientCPU' },
-    { name: 'Server - Nault Server', value: 'server' },
+    { name: 'Client Side - GPU/WebGL', value: 'clientWebGL' },
+    { name: 'Client Side - CPU (Slowest)', value: 'clientCPU' },
+    { name: 'Selected Server - dPoW / boomPoW / Other', value: 'server' },
```
Comment (BitDesert) [inline src/app/components/configure-app/configure-app.component.ts]: For consistency with "Client Side":
```suggestion
    { name: 'Server Side - dPoW / boomPoW / Other', value: 'server' },
```

Are the possible PoW mechanisms relevant "dPoW / boomPoW / Other" ?

File: src/app/components/configure-app/configure-app.component.ts [hunk 2]
```diff
@@ -300,12 +316,30 @@ export class ConfigureAppComponent implements OnInit {
...
+      if (await this.clearWorkCache()) {
+        reloadPending = true; // force reload balance => re-work pow
+      }
     }
 
     const newSettings = {
       walletStore: newStorage,
       lockInactivityMinutes: Number(this.selectedInactivityMinutes),
       powSource: newPoW,
+      multiplierSource: Number(this.selectedMultiplierOption),
```
Comment (BitDesert) [inline src/app/components/configure-app/configure-app.component.ts:342]: Why do we need to cast `this.selectedMultiplierOption` to Number? Isn't it already set to number in line 123?

**Review by BitDesert — CHANGES_REQUESTED** [review]

File: src/app/components/configure-app/configure-app.component.ts (hunk 2 above)
Comment (Joohansson) [reply src/app/components/configure-app/configure-app.component.ts:342]: Yeah that was strange. Without the Number, it got converted to string in the local store. Casting to Number solved the issue

File: src/app/components/configure-app/configure-app.component.ts (hunk 1 above)
Comment (Joohansson) [reply src/app/components/configure-app/configure-app.component.ts]: Solved

**Comment by keerifox** [comment]
small fix for remote PoW option name: https://github.com/keerifox/Nault/commit/018b49c56d2032ea913491bd5d773d411188ebe3

should be good to go otherwise

**Pull request merged**